scrapy/statscollectors.py

```diff
@@ -43,13 +43,16 @@ def open_spider(self, spider):
 
     def close_spider(self, spider, reason):
         if self._dump:
-            logger.info("Dumping Scrapy stats:\n" + pprint.pformat(self._stats),
+            logger.info("Dumping Scrapy stats:\n{}".format(self),
                         extra={'spider': spider})
         self._persist_stats(self._stats, spider)
 
     def _persist_stats(self, stats, spider):
         pass
 
+    def __format__(self, _):
```

Comment:
hm right, but I wouldn't know what to do with a format specifier for a statscollector. Any suggestions?